src/exchanger.js

```diff
@@ -95,129 +48,96 @@ function exchangeRandos (randos, callback) {
-    metrics.push(JSON.stringify({randoId: randos[i].randoId, mark: randos[i].mark}));
-  }
-  logger.info("[printMetrics]", "Metrics[chooser", chooser.randoId, "]:", metrics);
-}
-
-function printChooser (choosers) {
-  var printableChoosers = [];
-  for (var i = 0; i < choosers.length; i++) {
-    printableChoosers.push({chooserId: choosers[i].randoId, chooserEmail: choosers[i].user.email});
-  }
-  logger.info("[printChooser] Choosers:", printableChoosers);
-}
-
-function hasUserRando (rando, user) {
-  for (var i = 0; i < user.in.length; i++) {
-    if (user.in[i].randoId === rando.randoId) {
-      return true;
-    }
-  }
-  return false;
-}
-
-function findAllChoosers (randos) {
-  var choosers = [];
-  for (var i = 0; i < randos.length; i++) {
-    if (!isRandoWasChooser(randos[i], randos)) {
-      choosers.push(randos[i]);
-    }
-  }
-  return choosers;
-}
-
-function selectBestRando(randos) {
-  var bestRando = randos[0];
-  logger.trace("[exchanger.selectBestRando]", "Starting with best rando:", bestRando.randoId, "[", bestRando.mark, "]");
-  for (var i = 1; i < randos.length; i++) {
-    logger.trace("[exchanger.selectBestRando]", bestRando.randoId, "[", bestRando.mark,"] < ", randos[i].randoId, "[", randos[i].mark ,"]");
-    if (bestRando.mark < randos[i].mark) {
-      bestRando = randos[i];
-      logger.trace("[exchanger.selectBestRando]", "Update bestRando to ", randos[i].randoId);
-    }
-  }
-  return bestRando;
-}
-
 function putRandoToUserAsync (chooser, rando, randos, callback) {
   logger.trace("[exchanger.putRandoToUserAsync]", "Trying put rando to user in db");
   async.waterfall([
     function fetchUser (done) {
-      fetchUserByEmail(chooser.email, done);
+      dbService.fetchUserByEmail(chooser.email, done);
     },
     function putRandoToUserIn (user, done) {
       logger.trace("[exchanger.putRandoToUserAsync.putRandoToUserIn]", "Put rando to user.in");
-      logger.data("Rando", rando.randoId, "by", rando.user.email, "----in--->", user.email);
+      logger.data("Rando", rando.randoId, "by", rando.email, "----in--->", user.email);
       user.in.push(rando);
       done(null, user);
     },
     function updateUser (user, done) {
       logger.trace("[exchanger.putRandoToUserAsync.updateUser]", "Updating user");
-      updateUserOnRandos(user, randos);
       db.user.update(user, function (err) {
+        global.users[user.email] = user;
         done(err, user);
       });
     },
     function sendNotificationToUser (user, done) {
       logger.trace("[exchanger.sendNotificationToUser]", "Send message with ranodId: ", rando.randoId ," to user:", user.email);
       var message = {
         notificationType: "received",
-        rando: buildRando(rando)
+        rando: randoService.buildRando(rando)
       };
 
-      firebase.sendMessageToDevices(message, findActiveFirabseIds(user), done);
+      firebaseService.sendMessageToDevices(message, firebaseService.findActiveFirabseIds(user), done);
```

Comment:
It looks strange when we pass to service method result of execution of different method of same service. I think it's better pass user and call findActiveFirabseIds(user) privately.